library/std/src/sys/unix/fs.rs

```diff
@@ -332,11 +332,16 @@ pub struct FileTimes {
     modified: Option<SystemTime>,
 }
 
-#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+#[derive(Copy, Clone, Eq, Hash, Debug)]
```

Comment:
if we change equality behavior then we also need to change the `Hash` impl so that they're consistent.

Reply:
Oh, nice catch. But in this case, what would you think of rather implementing the invariant with a constructor? This would feel "cleaner" but the obvious disadvantage would be to have to add a module inside `unix:fs` to encapsulate `FileType`.

Reply:
Constructor sounds good. A separate module isn't necessary since this is an internal type that isn't exported. It's not a soundness issue after all, only a correctness one.